src/lib/logger.js

```diff
@@ -0,0 +1,143 @@
+   * completely breaks our system and requires a complete system reboot.
+   * On the client it will refresh the page after displaying an alert,
+   * on the server it will exit the server which will make forever restart
+   * the server.
+   * @param {Error | string} err - The fatal error to be logged, either a string or
+   * an Error object
+   */
+  fatal = async err => {
+    const errorObject = err instanceof Error ? err : new Error(err);
+    if (this.isClient) {
+      // We are actually lying currently, as developers haven't been notified
+      // when we hook up sentry we will be notifying developers though
+      await this.displayAlert(
+        'A fatal error has occurred, and the developers' +
+          ' have been notified. We apologize for the inconvenience, and we are afraid' +
+          ' that we will have to reload the page now',
+      );
+      window.location.reload(true);
+    } else {
+      // eslint-disable-next-line no-console
+      console.error(
+        'A fatal error occured, and the server is exiting. See the error below:',
+      );
+      // eslint-disable-next-line no-console
+      console.error(errorObject);
+      process.exit(1);
+    }
+  };
+
+  /**
+   * Logs an error. This should only be called on unexpected behaviour that should
+   * never happen as long as everything works
+   * @param {Error | string} err - The error to be logged, either a string or
+   * an Error object
+   */
+  error = err => {
+    const errorObject = err instanceof Error ? err : new Error(err);
+    // eslint-disable-next-line no-console
+    console.error(errorObject);
+  };
+
+  /**
+   * Logs a warning. This should be called when an event occurs that is usually unwanted
+   * but under certain circumstances could be okay
+   * @param {Error | string} warning - The warning to be logged, either a string or
+   * an Error object
+   */
+  warn = warning => {
+    const errorObject = warning instanceof Error ? warning : new Error(warning);
+    // eslint-disable-next-line no-console
+    console.warn(errorObject);
+  };
+
+  /**
+   * Logs a debug statement. This can be called whereever the developer thinks
+   * information could be helpful to have when debugging. It is the lowest logging
+   * level so will normally only be chosen to be shown in development mode
+   * @param {Error | string} msg - The warning to be logged, either a string or
+   * an Error object
+   */
+  debug = msg => {
+    const errorObject = msg instanceof Error ? msg : new Error(msg);
+    // eslint-disable-next-line no-console
+    console.log(errorObject);
+  };
+}
+
+let loggerInstance;
+
+/**
+ * @param {boolean} isClient - Whether we are currently on the client or the server
+ * @param {func} [displayAlert] - This argument displays an alert on the client side
+ * to the user, and should therefore only be supplied if isClient is true
+ */
+export const initializeLogger = (...args) => {
```

Comment:
Generally simple and neat implementation Emil! For our SWE class purposes those, I would propose that you make this logger follow the singleton pattern. That is, you give it a method `getLogger` which returns a static private variable `log` if it is set, otherwise it creates a new object. You can then add two more functions such as `setDisplayAlert` and `setClient`. I don't think this adds a lot of functionality, but it showcases that we have understood and know how to apply the pattern. LMKWYT.

Reply:
I think that might actually also be cleaner, I'll work on that, I was thinking that some of this would be pretty unclear especially to a beginner, so let's do it :)

Reply:
Sounds good. Let me know once you have done it, and I'll have a look :)